### PR TITLE
Adding BaseRequestBuilder.SetURL

### PR DESCRIPTION
**Pull request by DeepDiver1975**
This allows to specify an alternative BaseURL for the client.

### CODE REVIEW

File: gen/msgraph.go.tmpl [hunk 1]
```diff
@@ -102,6 +102,11 @@ func (r *BaseRequestBuilder) URL() string {
 	return r.baseURL
 }
 
+// SetURL returns URL
```
Comment (yaegashi) [inline gen/msgraph.go.tmpl]: Please update doc to reflect its functionality.

**Review by yaegashi — CHANGES_REQUESTED** [review]

**Comment by yaegashi** [comment]
@DeepDiver1975 Thanks for the PR!

I used to think of this kind of feature to enable custom baseURL.  An alternative would be adding a constructor method with another signature like `NewClientWithBaseURL(cli *http.Client, baseURL string)` and keeping baseURL immutable after creation.  Which is better?

Could you share some about the actual use cases you expect?  Part of the unit testing infrastructure?

**Comment by DeepDiver1975** [comment]
> Could you share some about the actual use cases you expect?

There are open source projects out there which implement the graph api and in this case the graph is accessible on different domains.

I'll update the PR asap. :+1:

**Comment by yaegashi** [comment]
Ok, after thiking a bit, I decided not to insist on the immutability of baseURL, so only adding SetURL() would suffice.  I will merge the PR after amending it.

> There are open source projects out there which implement the graph api and in this case the graph is accessible on different domains.

Nice to hear that msgraph.go has been somehow useful for projects by others. 😄

**Review by yaegashi — APPROVED** [review]

**Comment by DeepDiver1975** [comment]
> Nice to hear that msgraph.go has been somehow useful for projects by others. smile

Oh - absolutly useful! THX a lot for it!

**Pull request merged**